explorer.go

```diff
@@ -169,6 +161,34 @@ func (c *appContext) addressPage(w http.ResponseWriter, r *http.Request) {
 	io.WriteString(w, str)
 }
 
+func (c *appContext) search(w http.ResponseWriter, r *http.Request) {
+	searchStr := r.URL.Query().Get("search")
+	_, err := c.BlockData.GetBlockHeight(searchStr)
```

Comment:
As this is refined, let's please put plenty of comments within the function in addition to the docs above the function.  Specifically, talk through the logic of how you decide what the string represents.